README.md

```diff
@@ -1,2 +1,288 @@
+    The trapezoid which represents the body of the robot.
+
+    ![](https://i.imgur.com/hQBeLkl.png)
+
+    In the above Figure Leg extension is defined as the variable L is the vertical distance between the axis of the hip joints and the foot joint. The leg is shown at minimum extension Lmin the anti-parallel configuration and at maximal extension Lmax in the parallel configuration.
+
+    ![](https://i.imgur.com/lHossMi.png)
+
+    The above figure shown when the bot is at an arbitary position.
+    L1,L2,L3 are the length of linkages (System design parameters / variables).
+
+    <font color="green">
+    L1 ---> Length of Upper Leg Segment (From Hip Joint to Knee Joint)
+
+    L2 ---> Length of Lower Leg Segment (From Knee Joint to Foot Joint)
+
+    L3 --> Length from the Hip joint to the Mechanism's axis of Symmetry.
+    </font>
+
+* **Equations Used To Derive the System Design Variables**
+
+    ![](https://i.imgur.com/E1Gui6i.png)
+
+    In Above Image we have equations to calculate the <font color="yellow">$\phi$1, $\phi$2, L1, L2 and L3 </font>.
+
+    ![](https://i.imgur.com/1IjtgOg.png)
+
+    In above Image, it is the General Equation of Length (vertical distance between the axis of the hip joints and the foot joint)
+    Here, G (Gear Ratio) which is taken to be 1 for our Bot.
+    
+    
+    ![](https://i.imgur.com/9046wKp.png)
+
+    The above Graph gives is about change of Leg Extension wrt change in Driver Motor Angle. 
+
+
+    [Link To Python Script Used]()
+    [LInk to Python Script to Plot the Graph]()
+
+* **Specfications of Bot**
+    <font color="red">
+    
+    Lmax = 13 cm
+
+    Lmin = 4 cm
+    
+    L1 = 4.286 cm
+    
+    L2 = 9.236 cm
+
+    L3 = 3.5 cm
+
+    Phi1 = 45
+
+    Phi2 = 15
+
+    </font>
+
+
+
+
+## <ins> Components
+* Bot Parts of Acrylic Material (Laser-Cutted)
+* N20 motor Shaft for Wheels (3D Printed)
+* SRA PCB 
+* N20 motors
+* MG90 Servo Motors
+* Buck Converter (LM2596)
+* Bearings
+* Shafts
+
+> :memo: **Note:** [Link to Components used In Robot and Description](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/Components.md)
+
+## <ins> PID
+A PID controller is one kind of device used to control different process variables like pressure, flow, temperature, and speed in industrial applications. 
+
+> :memo: **Note:** [PID Notes](https://github.com/KudoC0nan/Balance_On_Wheels/blob/dev_arsh/Notes/PID%20Explanation)
+
+## <ins> Code Flow
+
+Here, we First wrote the algorithm for Self-Balancing the bot.
+
+
+For self-balancing the bot, we need to keep the bot a bit above our desired angle (Setpoint), since the bot is very fast and will fall down once it's past the desired angle.
```

Comment:
Also you run pid for balancing and lock servos simultaneously as well right otherwise your bot will collapse. mention that you set zero the servos first. mention both parts in your explanation. 

why only maximum to minimum? explain variable height and jump as well

Reply:
Added Explanations!